### ISSUE TITLE
Unit 7 mlagents-push-to-hf is not working

### ISSUE THREAD

**Issue by bantu-4879** [huggingface/deep-rl-class#475]
**mlagents-push-to-hf is not working** I have already created a Issue on mlagents repo. I have tried using this in two different environments. First in macOS, second in windows10. I have also tried in uninstalling and reinstalling mlagents but didn't worked. mlagents-learn is working fine in both environments and i have already trained my model but not able to upload it.

@simoninithomas is there any other way i can upload model? I don't know how and which files to upload manually. Please help

**Comment by simoninithomas** [comment]
Hey there 👋 what's the error you got?

**Comment by bantu-4879** [comment]
```
zsh:1: command not found: mlagents-push-to-hf
```
mlagents-learn is working fine

**Comment by simoninithomas** [comment]
Strange, can you check which version of MLAgents you used?

**Comment by bantu-4879** [comment]
I used **python 3.9.6** while using this code 
```
%cd ml-agents
%pip install -e ./ml-agents-envs
%pip install -e ./ml-agents
```
I got this error
```
[/Users/amankankriya/Documents/Deep](https://file+.vscode-resource.vscode-cdn.net/Users/amankankriya/Documents/Deep) RL/ml-agents
Obtaining file:///Users/amankankriya/Documents/Deep%20RL/ml-agents/ml-agents-envs
Requirement already satisfied: cloudpickle in [/Users/amankankriya/Documents/Deep](https://file+.vscode-resource.vscode-cdn.net/Users/amankankriya/Documents/Deep) RL/soccer/lib/python3.9/site-packages (from mlagents-envs==1.1.0.dev0) (3.0.0)
Collecting grpcio<=1.48.2,>=1.11.0
  Using cached grpcio-1.48.2.tar.gz (22.0 MB)
Requirement already satisfied: Pillow>=4.2.1 in [/Users/amankankriya/Documents/Deep](https://file+.vscode-resource.vscode-cdn.net/Users/amankankriya/Documents/Deep) RL/soccer/lib/python3.9/site-packages (from mlagents-envs==1.1.0.dev0) (10.2.0)
Requirement already satisfied: protobuf<3.20,>=3.6 in [/Users/amankankriya/Documents/Deep](https://file+.vscode-resource.vscode-cdn.net/Users/amankankriya/Documents/Deep) RL/soccer/lib/python3.9/site-packages (from mlagents-envs==1.1.0.dev0) (3.19.6)
Requirement already satisfied: pyyaml>=3.1.0 in [/Users/amankankriya/Documents/Deep](https://file+.vscode-resource.vscode-cdn.net/Users/amankankriya/Documents/Deep) RL/soccer/lib/python3.9/site-packages (from mlagents-envs==1.1.0.dev0) (6.0.1)
Requirement already satisfied: gym>=0.21.0 in [/Users/amankankriya/Documents/Deep](https://file+.vscode-resource.vscode-cdn.net/Users/amankankriya/Documents/Deep) RL/soccer/lib/python3.9/site-packages (from mlagents-envs==1.1.0.dev0) (0.26.2)
Requirement already satisfied: pettingzoo==1.15.0 in [/Users/amankankriya/Documents/Deep](https://file+.vscode-resource.vscode-cdn.net/Users/amankankriya/Documents/Deep) RL/soccer/lib/python3.9/site-packages (from mlagents-envs==1.1.0.dev0) (1.15.0)
Requirement already satisfied: numpy<1.24.0,>=1.21.2 in [/Users/amankankriya/Documents/Deep](https://file+.vscode-resource.vscode-cdn.net/Users/amankankriya/Documents/Deep) RL/soccer/lib/python3.9/site-packages (from mlagents-envs==1.1.0.dev0) (1.21.2)
Requirement already satisfied: filelock>=3.4.0 in [/Users/amankankriya/Documents/Deep](https://file+.vscode-resource.vscode-cdn.net/Users/amankankriya/Documents/Deep) RL/soccer/lib/python3.9/site-packages (from mlagents-envs==1.1.0.dev0) (3.13.1)
ERROR: Package 'mlagents-envs' requires a different Python: 3.9.6 not in '<=3.10.12,>=3.10.1'
WARNING: You are using pip version 21.1.3; however, version 23.3.2 is available.
You should consider upgrading via the '/Users/amankankriya/Documents/Deep RL/soccer/bin/python -m pip install --upgrade pip' command.
Note: you may need to restart the kernel to use updated packages.
Obtaining file:///Users/amankankriya/Documents/Deep%20RL/ml-agents/ml-agents
Collecting grpcio<=1.48.2,>=1.11.0
  Using cached grpcio-1.48.2.tar.gz (22.0 MB)
Requirement already satisfied: h5py>=2.9.0 in [/Users/amankankriya/Documents/Deep](https://file+.vscode-resource.vscode-cdn.net/Users/amankankriya/Documents/Deep) RL/soccer/lib/python3.9/site-packages (from mlagents==1.1.0.dev0) (3.10.0)
ERROR: Could not find a version that satisfies the requirement mlagents_envs==1.1.0.dev0 (from mlagents) (from versions: 0.10.0.dev1, 0.10.0, 0.10.1, 0.11.0.dev0, 0.11.0, 0.12.0, 0.12.1, 0.13.0, 0.13.1, 0.14.0, 0.14.1, 0.15.0, 0.15.1, 0.16.0, 0.16.1, 0.17.0, 0.18.0, 0.18.1, 0.19.0, 0.20.0, 0.21.0, 0.21.1, 0.22.0, 0.23.0, 0.24.0, 0.24.1, 0.25.0, 0.25.1, 0.26.0, 0.27.0, 0.28.0, 0.29.0, 0.30.0)
ERROR: No matching distribution found for mlagents_envs==1.1.0.dev0
WARNING: You are using pip version 21.1.3; however, version 23.3.2 is available.
You should consider upgrading via the '/Users/amankankriya/Documents/Deep RL/soccer/bin/python -m pip install --upgrade pip' command.
Note: you may need to restart the kernel to use updated packages.
```
Then i did ```pip3 install mlagents==0.30.0 mlagents-envs==0.30.0``` in same python environment and started training. Training is completed and while running this code ```!mlagents-push-to-hf --run-id="SoccerTwos1"  --local-dir="./results/SoccerTwos1/SoccerTwos"  --repo-id="Amankankriya/poca-SoccerTwos" --commit-message="First Push"``` I faced this error ```zsh:1: command not found: mlagents-push-to-hf```



When i try same code in **python 3.10.11**, ```mlagents-envs``` successfully installed but ```mlagents``` is showing this error (_given i already installed cmake, protobuf; I tried to install onnx==1.12.0 but could not complete with same legacy error_)
```× Running setup.py install for onnx did not run successfully.
  │ exit code: 1
  ╰─> [89 lines of output]
      fatal: not a git repository (or any of the parent directories): .git
      [/Users/amankankriya/Documents/Deep](https://file+.vscode-resource.vscode-cdn.net/Users/amankankriya/Documents/Deep) RL/amanb/lib/python3.10/site-packages/setuptools/installer.py:27: SetuptoolsDeprecationWarning: setuptools.installer is deprecated. Requirements should be satisfied by a PEP 517 installer.
        warnings.warn(
      running install
      [/Users/amankankriya/Documents/Deep](https://file+.vscode-resource.vscode-cdn.net/Users/amankankriya/Documents/Deep) RL/amanb/lib/python3.10/site-packages/setuptools/command/install.py:34: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.
        warnings.warn(
      running build
      running build_py
      running create_version
      running cmake_build
      Using cmake args: ['/Users/amankankriya/Documents/Deep RL/amanb/bin/cmake', '-DPYTHON_INCLUDE_DIR=/Library/Frameworks/Python.framework/Versions/3.10/include/python3.10', '-DPYTHON_EXECUTABLE=/Users/amankankriya/Documents/Deep RL/amanb/bin/python', '-DBUILD_ONNX_PYTHON=ON', '-DCMAKE_EXPORT_COMPILE_COMMANDS=ON', '-DONNX_NAMESPACE=onnx', '-DPY_EXT_SUFFIX=.cpython-310-darwin.so', '-DCMAKE_BUILD_TYPE=Release', '-DONNX_ML=1', '/private/var/folders/54/9xphnqpn00b76nztg29d2t8w0000gn/T/pip-install-wc9nswpi/onnx_292111e7f77b453595da3d8d1d57e0c2']
      CMake Deprecation Warning at CMakeLists.txt:2 (cmake_minimum_required):
        Compatibility with CMake < 3.5 will be removed from a future version of
        CMake.
      
        Update the VERSION argument <min> value or use a ...<max> suffix to tell
        CMake that the project does not need compatibility with older versions.
      
      
      CMake Warning (dev) at CMakeLists.txt:117 (find_package):
        Policy CMP0148 is not set: The FindPythonInterp and FindPythonLibs modules
        are removed.  Run "cmake --help-policy CMP0148" for policy details.  Use
        the cmake_policy command to set the policy and suppress this warning.
      
      This warning is for project developers.  Use -Wno-dev to suppress it.
      
      CMake Warning (dev) at CMakeLists.txt:118 (find_package):
        Policy CMP0148 is not set: The FindPythonInterp and FindPythonLibs modules
        are removed.  Run "cmake --help-policy CMP0148" for policy details.  Use
        the cmake_policy command to set the policy and suppress this warning.
      
      This warning is for project developers.  Use -Wno-dev to suppress it.
      
      Generated: [/private/var/folders/54/9xphnqpn00b76nztg29d2t8w0000gn/T/pip-install-wc9nswpi/onnx_292111e7f77b453595da3d8d1d57e0c2/.setuptools-cmake-build/onnx/onnx-ml.proto](https://file+.vscode-resource.vscode-cdn.net/private/var/folders/54/9xphnqpn00b76nztg29d2t8w0000gn/T/pip-install-wc9nswpi/onnx_292111e7f77b453595da3d8d1d57e0c2/.setuptools-cmake-build/onnx/onnx-ml.proto)
      CMake Error at CMakeLists.txt:299 (message):
        Protobuf compiler not found
      Call Stack (most recent call first):
        CMakeLists.txt:330 (relative_protobuf_generate_cpp)
      
      
      -- Configuring incomplete, errors occurred!
      Traceback (most recent call last):
        File "<string>", line 2, in <module>
        File "<pip-setuptools-caller>", line 34, in <module>
        File "/private/var/folders/54/9xphnqpn00b76nztg29d2t8w0000gn/T/pip-install-wc9nswpi/onnx_292111e7f77b453595da3d8d1d57e0c2/setup.py", line 332, in <module>
          setuptools.setup(
    ...

× Encountered error while trying to install package.
╰─> onnx```

**Comment by simoninithomas** [comment]
Are you cloning from: https://github.com/Unity-Technologies/ml-agents/ ?

**Comment by bantu-4879** [comment]
At first yes but when it didn't work i tried ```pip3 install mlagents==0.30.0 mlagents-envs==0.30.0```

**Comment by Yusuf-Agac** [comment]
I have same error :(


(base) PS C:\Users\Yusuf\Documents\GitHub\AircraftAI> mlagents-push-to-hf --run-id="SoccerTwos" --local-dir="./results/SoccerTwos" --repo-id="YusufTree/poca-SoccerTwos" --commit-message="First Push"
mlagents-push-to-hf : The term 'mlagents-push-to-hf' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spelling of the name, or if 
a path was included, verify that the path is correct and try again.
At line:1 char:1
+ mlagents-push-to-hf --run-id="SoccerTwos" --local-dir="./results/Socc ...
+ ~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (mlagents-push-to-hf:String) [], CommandNotFoundException
    + FullyQualifiedErrorId : CommandNotFoundException


![image](https://github.com/huggingface/deep-rl-class/assets/74566648/a27c7d06-f18a-46ed-b27f-5cf31aff42f7)

**Comment by simoninithomas** [comment]
@Yusuf-Agac that's weird 🤔. What was the message during the mlagents setup?

**Comment by Yusuf-Agac** [comment]
@simoninithomas A while ago, I installed the mlagents package on my local computer by looking at the Unity documentation. So I don't remember the message. However, there were no problems in training. When I couldn't solve the problem, I uploaded the model by dragging a file from the hugging face interface.

By the way, it is truly one of the best prepared courses I have completed in my life. Thanks for the course. 💯
